x/archivedb/db_test.go

```diff
@@ -0,0 +1,170 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package archivedb
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/database/memdb"
+)
```

Comment:
It would be great if we could hook up the batch to the relevant tests [here](https://github.com/ava-labs/avalanchego/blob/b820b46ee8d51ab1ae05da3294ff941e6f901137/database/test_database.go#L239) in `database/test_database.go`